doc/api/benchmark.md

```diff
@@ -0,0 +1,293 @@
+# Benchmark
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1.1 - Active Development
+
+<!-- source_link=lib/benchmark.js -->
+
+The `node:benchmark` module gives the ability to measure
+performance of JavaScript code. To access it:
+
+```mjs
+import benchmark from 'node:benchmark';
+```
+
+```cjs
+const benchmark = require('node:benchmark');
+```
+
+This module is only available under the `node:` scheme. The following will not
+work:
+
+```mjs
+import benchmark from 'benchmark';
+```
+
+```cjs
+const benchmark = require('benchmark');
+```
+
+The following example illustrates how benchmarks are written using the
+`benchmark` module.
+
+```mjs
+import { Suite } from 'node:benchmark';
+
+const suite = new Suite();
+
+suite.add('Using delete to remove property from object', function() {
+  const data = { x: 1, y: 2, z: 3 };
+  delete data.y;
+
+  data.x;
+  data.y;
+  data.z;
+});
+
+suite.run();
+```
+
+```cjs
+const { Suite } = require('node:benchmark');
+
+const suite = new Suite();
+
+suite.add('Using delete to remove property from object', function() {
+  const data = { x: 1, y: 2, z: 3 };
+  delete data.y;
+
+  data.x;
+  data.y;
+  data.z;
+});
+
+suite.run();
+```
+
+```console
+$ node my-benchmark.js
+(node:14165) ExperimentalWarning: The benchmark module is an experimental feature and might change at any time
+(Use `node --trace-warnings ...` to show where the warning was created)
+Using delete property x 5,853,505 ops/sec ± 0.01% (10 runs sampled)     min..max=(169ns ... 171ns) p75=170ns p99=171ns
+```
+
+## Class: `Suite`
+
+> Stability: 1.1 Active Development
+
+<!-- YAML
+added: REPLACEME
+-->
+
+An `Suite` is responsible for managing and executing
+benchmark functions. It provides two methods: `add()` and `run()`.
+
+### `new Suite([options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `options` {Object} Configuration options for the suite. The following
+  properties are supported:
+  * `reporter` {Function} Callback function with results to be called after
+    benchmark is concluded. The callback function should receive two arguments:
+    `suite` - A {Suite} object and
+    `result` - A object containing three properties:
+    `opsSec` {string}, `iterations {Number}`, `histogram` {Histogram} instance.
```

Comment:
For Node.js' own benchmarks we have the ability to specify the number of iterations for the the run and finer grained control over exactly when to start and stop measuring. Ideally we'd be able to implement Node.js' own benchmarks in terms of this api so I'm just curious how you imagine those bits coming together here.

Reply:
oh, ha! I guess I should have read further down first! I see you already have iterations there!

Reply:
We didn't have control over how many iterations you can run but you have control over how much time you want your benchmark to run.

The `.count` property on `timer` is basically the number of times you should run your benchmark function to achieve `bench.minTime`, but you can ignore and just specify how many iterations do you want (even if is not built for that)